File: sdks/go/calendar.go
package aegis

// CalendarService handles operations on the /api/calendar endpoints.
type CalendarService struct {
	client *Client
}

// Events returns all calendar events for the given month (format "YYYY-MM"),
// including projected recurring entries.
func (s *CalendarService) Events(month string) ([]CalendarEvent, error) {
	qs := queryString(map[string]string{"month": month})
	var events []CalendarEvent
	err := s.client.get("/api/calendar/events"+qs, &events)
	return events, err
}

// Summary returns the calendar summary for the given month (format "YYYY-MM").
func (s *CalendarService) Summary(month string) (*CalendarSummary, error) {
	qs := queryString(map[string]string{"month": month})
	var summary CalendarSummary
	err := s.client.get("/api/calendar/summary"+qs, &summary)
	return &summary, err
}

File: sdks/typescript/src/resources/calendar.ts
import type { AegisClient } from "../client";
import type { CalendarEvent, CalendarSummary } from "../types";

export class CalendarResource {
  private client: AegisClient;

  constructor(client: AegisClient) {
    this.client = client;
  }

  /** Get calendar events for a given month (format: "YYYY-MM"). */
  async getEvents(month: string): Promise<CalendarEvent[]> {
    return this.client.get<CalendarEvent[]>("/api/calendar/events", { month });
  }

  /** Get a calendar summary for a given month (format: "YYYY-MM"). */
  async getSummary(month: string): Promise<CalendarSummary> {
    return this.client.get<CalendarSummary>("/api/calendar/summary", { month });
  }
}
